### PR TITLE
fix: only fetch tutorials map once

### PR DESCRIPTION
**Pull request by dstaley**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](url) 🎟️

## 🗒️ What

This PR refactors the `getStaticProps` implementation for `docs-view` to only generate/fetch the tutorials map once. In a future PR, I'm going to explore if we even need to fetch the tutorials map to rewrite URLs anymore.

## 🤷 Why

Prior to this PR, `getStaticProps` would generate/fetch the tutorials map multiple times, each taking ~2 seconds. This was because different steps of the implementation would fetch their own copy of the tutorials map independent of one another.

## 🛠️ How

In this PR, we fetch the tutorials map once at the beginning of `getStaticProps`, and pass it into the remote content loader (for link rewrites) and again into the `prepareNavData` utility.

## 🧪 Testing

- [ ] Click through a handful of docs pages and ensure that they render.
- [ ] Go to [preview](https://dev-portal-git-dstutorials-map-once-hashicorp.vercel.app/terraform/intro) and ensure the "Getting Started" link in the sidebar does not point to `learn.hashicorp.com`.

### CODE REVIEW

File: src/lib/learn-client/index.ts [hunk 1]
```diff
@@ -2,19 +2,6 @@
  * Copyright (c) HashiCorp, Inc.
  * SPDX-License-Identifier: MPL-2.0
  */
-function getFetch() {
```
Comment (dstaley) [inline src/lib/learn-client/index.ts:5]: This was removed in favor of native `fetch` to (hopefully) improve accuracy of Vercel's "Request Metrics" feature.

File: src/lib/remark-plugins/rewrite-tutorial-links/utils/get-tutorial-map.ts [hunk 2]
```diff
@@ -3,7 +3,6 @@
  * SPDX-License-Identifier: MPL-2.0
  */
 
-import fetch from 'node-fetch'
```
Comment (dstaley) [inline src/lib/remark-plugins/rewrite-tutorial-links/utils/get-tutorial-map.ts:6]: Removing to use the native global `fetch` for better integration with Vercel.

**Comment by mwickett** [comment]
nice!

**Comment by dstaley** [comment]
@heatlikeheatwave I think we can replace this async request to build the tutorials map with one statically generated prior to build. What do you think about that approach (in combination with this one where we pass the values instead of regenerating them).

File: src/layouts/sidebar-sidecar/utils/prepare-nav-data-for-client.ts
Comment (heatlikeheatwave) [inline src/layouts/sidebar-sidecar/utils/prepare-nav-data-for-client.ts:1]: Looking at this file is giving me take home coding challenge flashbacks 😂

Comment (dstaley) [reply src/layouts/sidebar-sidecar/utils/prepare-nav-data-for-client.ts:1]: fun fact: that's exactly where that component of the take home came from! (well the original component that was used for the same purpose on the .io sites)

**Review by heatlikeheatwave — APPROVED** [review]

File: src/layouts/sidebar-sidecar/utils/prepare-nav-data-for-client.ts
Comment (heatlikeheatwave) [reply src/layouts/sidebar-sidecar/utils/prepare-nav-data-for-client.ts:1]: !!!

**Comment by heatlikeheatwave** [comment]
> @heatlikeheatwave I think we can replace this async request to build the tutorials map with one statically generated prior to build. What do you think about that approach (in combination with this one where we pass the values instead of regenerating them).

I think this approach (generate once, pass to other functions) makes sense to me! Tested it locally (in addition to the preview) and it looks good 👍

**Pull request merged**